apps/cli/src/exec/cartesi-machine.ts

```diff
@@ -0,0 +1,29 @@
+import { parse, SemVer } from "semver";
+import { execaDockerFallback, ExecaOptionsDockerFallback } from "./util";
+
+export const boot = async (
+    args: readonly string[],
+    options: ExecaOptionsDockerFallback,
+) => {
+    return execaDockerFallback("cartesi-machine", args, options);
+};
+
+export const version = async (options?: {
+    image?: string;
+}): Promise<SemVer | null> => {
```

Comment:
Can we have a `REQUIRED_VERSION` for each `exec/<module>` and if `<module>.version` doesn't match the required onde we throw a typed exception.

This way we could test the semver using minor or patch when possible, without needing an exact match in case it works for the `<module>` being used.

Reply:
you mean before the execution of the command?

Reply:
Not necessarily,  but just that the version required definition is part of the
<module> code and not the test code.

Reply:
I didn't understand your request

Reply:
something in those lines:

```patch
diff --git a/apps/cli/src/exec/crane.ts b/apps/cli/src/exec/crane.ts
index 15a402e..13c1bcf 100644
--- a/apps/cli/src/exec/crane.ts
+++ b/apps/cli/src/exec/crane.ts
@@ -2,6 +2,8 @@ import { parse, SemVer } from "semver";
 import { Stream } from "stream";
 import { spawnSyncDockerFallback } from "./util";

+export const RequiredVersion: SemVer = parse("0.19.1")!;
+
 export const exportImage = async (options: {
     stdin: Stream;
     stdout: Stream;
diff --git a/apps/cli/test/exec/crane.test.ts b/apps/cli/test/exec/crane.test.ts
index 50021ff..3638854 100644
--- a/apps/cli/test/exec/crane.test.ts
+++ b/apps/cli/test/exec/crane.test.ts
@@ -4,6 +4,6 @@ import { crane } from "../../src/exec/index.js";
 describe("crane", () => {
     it("should report version", async () => {
         const version = await crane.version({ image: "cartesi/sdk:0.11.0" });
-        expect(version?.format()).toEqual("0.19.1");
+        expect(version?.format()).toEqual(crane.RequiredVersion.format());
     });
 });
```

Reply:
done